ext/spl/tests/bug65069.phpt

```diff
@@ -0,0 +1,60 @@
+--TEST--
+Bug #65069: GlobIterator fails to access files inside an open_basedir restricted dir
+--FILE--
+<?php
+$file_path = dirname(__FILE__);
+// temp dirname used here
+$dirname = "$file_path/bug65069";
+// temp dir created
+mkdir($dirname);
+
+ini_set('open_basedir', $dirname);
+
+// temp files created
+$fp = fopen("$dirname/wonder12345", "w");
+fclose($fp);
+$fp = fopen("$dirname/wonder.txt", "w");
+fclose($fp);
+$fp = fopen("$dirname/file.text", "w");
+fclose($fp);
+
+$spl_glob_it = new \GlobIterator("$dirname/*.txt");
+foreach ($spl_glob_it as $file_info) {
+    echo $file_info->getFilename() . "\n";
+}
+
+$spl_glob_it = new \GlobIterator(dirname(dirname($dirname)) . "/*/*/*");
+foreach ($spl_glob_it as $file_info) {
+    echo $file_info->getFilename() . "\n";
+}
+
+$spl_glob_empty = new \GlobIterator("$dirname/*.php");
+var_dump($spl_glob_empty->count());
+
+// top directory
+var_dump(iterator_to_array(new \GlobIterator(dirname(dirname($dirname)))));
+
+// not existing file
+var_dump(iterator_to_array(new \GlobIterator("$file_path/bug65069-this-will-never-exists")));
+
+
+?>
+--CLEAN--
+<?php
+$file_path = dirname(__FILE__);
+$dirname = "$file_path/bug65069";
+unlink("$dirname/wonder12345");
+unlink("$dirname/wonder.txt");
+unlink("$dirname/file.text");
+rmdir($dirname);
+?>
+--EXPECT--
+wonder.txt
+file.text
+wonder.txt
```

Comment:
It might just be me being confused, but why does the ``wonder.txt`` file appear twice here?

Reply:
The first one is from the first iteration - `$dirname/*.txt"` and the second one is from `dirname(dirname($dirname)) . "/*/*/*"`

Reply:
Ah right :+1: